nidmviewerfsl/callSlicer.py

```diff
@@ -111,6 +120,7 @@ def resizeSPMtoFSL(exc_set, template, scalefactor, tempDir):
     
     #Run the command  if necessary.
     resizeCommand = "flirt -init " + os.path.join(tempDir, "resizeMatrix.mat") + " -in " + exc_set + " -ref " + template + " -out " + os.path.join(tempDir, "resizedNifti.nii.gz") + " -applyxfm"
+    subprocess.check_call(resizeCommand, shell=True)
```

Comment:
Do you need `check_call`+ `Popen`+ `output`? Is this running the command twice? Could we instead used `check_output`?

Reply:
I am having trouble using check_output. A minimal example of the issue would be the following:

~~~~
#Make the commands
    getDimString1 = ["fslhd", niftiFilename]
    getDimString2 = ["cat", "-v"]
    
    #Run the command
    process_1 = subprocess.Popen(getDimString1, shell=False, stdout=subprocess.PIPE)
    print(process_1.stdout)
    process_2 = subprocess.Popen(getDimString2, shell=False, stdin=process_1.stdout, stdout=subprocess.PIPE)
~~~~

The above function assigns the variable `process_1.stdout` to a buffered reader object which stores the output of the command `getDimString1`. The print statement gives the following; `<_io.BufferedReader name=6>`. This buffered reader object can then be read as input for the next command.

~~~~
process_1_stdout = subprocess.check_output(getDimString1, shell=False)
print(process_1_stdout)
process_2_stdout = subprocess.check_output(getDimString2, shell=False, stdin=process_1_stdout)
~~~~

However, if I try the same thing with checkoutput, whilst it does give me the output it doesn't give it as a buffered reader object. Instead it assigns it as a string. 

(e.g. the above prints `b'filename       /home/tom/Documents/Repos/nidmresults-fslhtml/Tests/data/ex_spm_HRF_informed_basis_test/ExcursionSet.nii.gz\n\nsizeof_hdr     348\ndata_type      FLOAT32\ndim0           3\ndim1           79\ndim2           95\ndim3           79\ndim4           1\ndim5           1\ndim6           1\ndim7           1\nvox_units      mm\ntime_units     s\ndatatype       16\nnbyper         4\nbitpix         32\npixdim0        0.000000\npixdim1        2.000000\npixdim2        2.000000\npixdim3        2.000000\npixdim4        0.000000\npixdim5        0.000000\npixdim6        0.000000\npixdim7        0.000000\nvox_offset     352\ncal_max        0.0000\ncal_min        0.0000\nscl_slope      1.000000\nscl_inter      0.000000\nphase_dim      0\nfreq_dim       0\nslice_dim      0\nslice_name     Unknown\nslice_code     0\nslice_start    0\nslice_end      0\nslice_duration 0.000000\ntime_offset    0.000000\nintent         Unknown\nintent_code    0\nintent_name    \nintent_p1      0.000000\nintent_p2      0.000000\nintent_p3      0.000000\nqform_name     Aligned Anat\nqform_code     2\nqto_xyz:1      -2.000000  0.000000  -0.000000  78.000000\nqto_xyz:2      0.000000  2.000000  -0.000000  -112.000000\nqto_xyz:3      0.000000  0.000000  2.000000  -70.000000\nqto_xyz:4      0.000000  0.000000  0.000000  1.000000\nqform_xorient  Right-to-Left\nqform_yorient  Posterior-to-Anterior\nqform_zorient  Inferior-to-Superior\nsform_name     Aligned Anat\nsform_code     2\nsto_xyz:1      -2.000000  0.000000  0.000000  78.000000\nsto_xyz:2      0.000000  2.000000  0.000000  -112.000000\nsto_xyz:3      0.000000  0.000000  2.000000  -70.000000\nsto_xyz:4      0.000000  0.000000  0.000000  1.000000\nsform_xorient  Right-to-Left\nsform_yorient  Posterior-to-Anterior\nsform_zorient  Inferior-to-Superior\nfile_type      NIFTI-1+\nfile_code      1\ndescrip        \naux_file       \n'`

This means I cannot use it as input for the next command. I'm not entirely sure why is happens as looking at the `check_output` code it looks like `checkout_output` should return a buffered reader object similar to `process_1.stdout` in the first block of code. But it doesn't. 

I can't seem to create a buffered reader object from this string either. I'm not sure how to work around this. What do you think?

I do agree though that `check_call` I think is just a repetition of what is already being done by `Popen` and `wait` as it appears that it is built using just these functions. Perhaps I should remove the `check_call`'s?

Reply:
I like to use `check_call` because it raises an error if the command returned an error message  but I don't think we should spend more time on this. In the future, we can either use `nibabel` or avoid the reslicing by displaying diretly on the corresponding template. How does this sound?

In the meantime, we should avoid to call twice the same command. Can you update the code to remove `check_call`?

Reply:
Okay, I shall remove the `check_call` but I do see what you mean about it being a shame we won't have the error message. Yes, I believe displaying on the SPM template removes a lot of these issues. If you like I can change it to the SPM template this afternoon (this will be a relatively quick fix I believe). Do you know where I could find the 1mm and 2mm MNI SPM templates?